cmd/translatesfx/translatesfx/directive.go

```diff
@@ -0,0 +1,138 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translatesfx
+
+import (
+	"fmt"
+	"strings"
+)
+
+type source int
+
+const (
+	directiveSourceUnknown source = iota
+	directiveSourceFile
+	directiveSourceEnv
+	directiveSourceZK
+	directiveSourceEtcd2
+	directiveSourceConsul
+	directiveSourceVault
+)
+
+type directive struct {
+	fromPath    string
+	defaultV    string
+	fromType    source
+	isDirective bool
```

Comment:
I'm curious why `isDirective` is preferable to a new interface and checked type assertions.  Is to avoid needing accessors?

Reply:
Yeah, that was a mistake. I removed `isDirective` in the PR I'm working on now :-)